front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts

```diff
@@ -53,24 +53,19 @@ async function handler(
     return apiError(req, res, keyRes.error);
   }
 
-  const { workspaceAuth, keyAuth } = await Authenticator.fromKey(
-    keyRes.value,
-    req.query.wId as string
-  );
-
-  if (
-    !workspaceAuth.isBuilder() ||
-    keyAuth.getNonNullableWorkspace().sId !== req.query.wId
-  ) {
+  const { wId } = req.query;
+  if (typeof wId !== "string") {
     return apiError(req, res, {
```

Comment:
You removed the check on workspaceAuth.isBuilder()

Which means Anyone from any workspace can tap into that endpoint :/

Reply:
I put back the check, but I'm not sure to see how this would be the case as workspaceAuth is dependant on workspaceId ?

Reply:
Anyone can possess an auth object for a specific workspace, even if they are not members of that workspace. In the context of the public API, this implies that all keys have the potential to access any workspace and will ultimately receive an auth object. The only way to verify the membership or permissions of the key/user is by checking their role (isUser/isBuilder/isAdmin).